### PR TITLE
Fix flaky tests in PortalRegistryTest.java and ConsulRegistryTest.java

### PR DESCRIPTION
**Pull request by KellyShao**
Fixed flaky test that caused by unregistered and unavailabled `registry` in class `PortalRegistryTest.java` and `ConsulRegistryTest.java`. 

- **Issue 1** 
    - In both classes, if only run `subAndUnsubService()` method before `doRegisterAndAvailable()` or `discoverService()` method, the tests will failed. Take the `PortalRegistryTest` class for example, the error msgs are:
        - `doRegisterAndAvailable()` method: 
            ```
              java.lang.AssertionError
                at com.networknt.portal.registry.PortalRegistryTest.doRegisterAndAvailable(PortalRegistryTest.java:76)
            ```
        - `discoverService()` method: 
            ```
             java.lang.AssertionError
               at com.networknt.portal.registry.PortalRegistryTest.a1discoverService(PortalRegistryTest.java:122)
            ```
    - **Root cause**: Since the `subAndUnsubService()` method is the last one, in the default test order in both `PortalRegistryTest.java` and `ConsulRegistryTest.java`. If we simply run the tests, we cannot find out the `subAndUnsubService()` method has unregistered URLs.

    - **Solution**: Unregistered URLs in the `subAndUnsubService()` method

- **Issue 2** 
    - In both classes, if only run `discoverService()` method before `doRegisterAndAvailable()` method, the tests will failed. Take the `PortalRegistryTest` class for example, the error msg is:
    ```
     java.lang.AssertionError
        at com.networknt.portal.registry.PortalRegistryTest.doRegisterAndAvailable(PortalRegistryTest.java:76)
     ```
    - **Root cause**: Since the `discoverService()` method is always after the `doRegisterAndAvailable()` method, in the default test order in both `PortalRegistryTest.java` and `ConsulRegistryTest.java`. If we simply run the tests, we cannot find out the `discoverService()` method has unavailabled service.

    - **Solution**: Unavailabled service in the `discoverService()` method
  
The flaky tests are found by running [iDFlakies](https://github.com/idflakies/iDFlakies), after fixing the issue, there's no flaky tests in `PortalRegistryTest.java` and `ConsulRegistryTest.java`.

### CODE REVIEW

File: portal-registry/src/test/java/com/networknt/portal/registry/PortalRegistryTest.java [hunk 1]
```diff
@@ -88,15 +88,23 @@ public void doRegisterAndAvailable() throws Exception {
         Assert.assertFalse(client.isRegistered(service2));
     }
 
+    @Ignore
```
Comment (KellyShao) [inline portal-registry/src/test/java/com/networknt/portal/registry/PortalRegistryTest.java:91]: Since the subscribe related code was commented in https://github.com/networknt/light-4j/commit/f8b87acef5f9738dee2e611f184619ec46e507fb, the current code in this class is not related to `subAndUnsubService`. I suggest simply **ignore** this class for now.

Comment (stevehu) [reply portal-registry/src/test/java/com/networknt/portal/registry/PortalRegistryTest.java:91]: Yes. Agreed.

**Comment by stevehu** [comment]
[Fix flaky tests in PortalRegistryTest.java and ConsulRegistryTest.java light-4j master](https://trello.com/c/hgw5QwSF/883-fix-flaky-tests-in-portalregistrytestjava-and-consulregistrytestjava-light-4j-master)

**Comment by stevehu** [comment]
@KellyShao Thanks a lot for your help.

File: portal-registry/src/test/java/com/networknt/portal/registry/PortalRegistryTest.java [hunk 2]
```diff
@@ -117,6 +125,11 @@ public void discoverService() throws Exception {
             e.printStackTrace();
         }
         Assert.assertTrue(urls.contains(serviceUrl));
+        
+        // unavailable
+        registry.doUnavailable(null);
+        Thread.sleep(sleepTime);
+        Assert.assertFalse(client.isWorking(serviceid));
```
Comment (BalloonWen) [inline portal-registry/src/test/java/com/networknt/portal/registry/PortalRegistryTest.java:132]: where is the serviceid defined?

File: portal-registry/src/test/java/com/networknt/portal/registry/PortalRegistryTest.java (hunk 1 above)
Comment (BalloonWen) [reply portal-registry/src/test/java/com/networknt/portal/registry/PortalRegistryTest.java:91]: the ignore is not imported

**Review by BalloonWen — COMMENTED** [review]
I think this PR fails the build.

File: portal-registry/src/test/java/com/networknt/portal/registry/PortalRegistryTest.java (hunk 1 above)
Comment (KellyShao) [reply portal-registry/src/test/java/com/networknt/portal/registry/PortalRegistryTest.java:91]: Added in https://github.com/networknt/light-4j/pull/862. Thanks!
